instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -212,7 +224,17 @@ public Connection connect(String url, Properties info) throws SQLException {
 
     DbInfo dbInfo = JdbcConnectionUrlParser.parse(realUrl, info);
 
-    return new OpenTelemetryConnection(connection, dbInfo, statementInstrumenter());
+    OpenTelemetry otel = getOpenTelemetry();
+    Instrumenter<DbRequest, Void> statementInstrumenter =
+        JdbcInstrumenterFactory.createStatementInstrumenter(otel);
+    return new OpenTelemetryConnection(connection, dbInfo, statementInstrumenter);
+  }
+
+  private static OpenTelemetry getOpenTelemetry() {
+    if (openTelemetry == null) {
+      return GlobalOpenTelemetry.get();
```

Comment:
Not sure it would be the expected behavior. This piece of code preserves the previous behavior.

Reply:
we dropped GlobalOpenTelemetry support from logging appenders: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8791#discussion_r1246965249

I support doing the same here.

(maybe related to #8080? cc @michalvavrik @brunobat)

Reply:
I have removed the GlobalOpenTelemetry usage

Reply:
> I don't agree with returning a `GlobalOpenTelemetry.get()` How do you make sure `connect()` is not called before `install()`?

@brunobat For Spring Boot, see: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ed5190a10c72ad283b96f056bff33088bf6b8546/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/jdbc/OpenTelemetryJdbcDriverAutoConfiguration.java#L33

Reply:
The Readme of the JDBC instrumentation library is also updated to explain that `OpenTelemetry` has to be injected into `OpenTelemetryDriver` before the initialization of the connection pool.